client/client.go

```diff
@@ -173,30 +173,99 @@ func unmarshalRepoPackages(p, cacheDir string, cacheLife time.Duration, proxySer
 	}
 	logger.Infof("Fetching repo content for %s, cache either doesn't exist or is older than %v", p, cacheLife)
 
-	url := p + "/index.gz"
-	logger.Infof("Fetching %q", url)
-	res, err := httpClient.Get(url)
+	parsedP, err := url.Parse(p)
 	if err != nil {
 		return nil, err
 	}
+	switch parsedP.Scheme {
+	case "http", "https":
```

Comment:
again, this should figure out if the repo is a gcs path based on parsing the URL